### PR TITLE
389-ds-base: 2.0.7 -> 2.3.0

### PR DESCRIPTION
**Pull request by ners**
###### Description of changes

Update the 389 Directory Server to the latest version. I followed the Fedora packaging instructiors here:
https://download-ib01.fedoraproject.org/pub/fedora/linux/development/rawhide/Everything/source/tree/Packages/3/389-ds-base-2.3.0-2.fc38.src.rpm

Of note is that I haven't actually managed to run the binary successfully, as the server seems to crash instantly without printing any output.

Also of note is the fact that the old derivation doesn't even build, so I can't check if it ran any better.

I also moved apache-directory-server to the ldap subdirectory.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/389-directory-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    openssl
+    zlib
+  ]
+  ++ lib.optional withSystemd systemd
+  ++ lib.optional withOpenldap openldap
+  ++ lib.optional withBdb db
+  ++ lib.optional withNetSnmp net-snmp;
+
+  postPatch = ''
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/servers/389-directory-server/default.nix:81]: Can we reduce the amount of patched files?

File: pkgs/servers/389-directory-server/default.nix [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  '';
+
+  preConfigure = ''
+    ./autogen.sh --prefix="$out"
+  '';
+
+  preBuild = ''
+    ln -s ${cargoDeps} /build/source/vendor
```
Comment (SuperSandro2000) [inline pkgs/servers/389-directory-server/default.nix:89]: Please use relative directories and never hardcode /build/

File: pkgs/servers/389-directory-server/default.nix [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  '';
+
+  configureFlags =
+    [
```
Comment (SuperSandro2000) [inline pkgs/servers/389-directory-server/default.nix:93]: We can drop the new line here to reduce indentation.

File: pkgs/servers/389-directory-server/default.nix [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    ] ++ lib.optionals withBdb [
+      "--with-db-inc=${db.dev}/include"
+      "--with-db-lib=${db.out}/lib"
+    ] ++ lib.optionals withNetSnmp [
+      "--with-netsnmp-inc=${net-snmp.dev}/include"
+      "--with-netsnmp-lib=${net-snmp.lib}/lib"
+    ];
+
+  meta = with lib; {
+    description = "An enterprise-class LDAP server for Linux developed by Red Hat as part of the community-supported Fedora Project";
```
Comment (SuperSandro2000) [inline pkgs/servers/389-directory-server/default.nix:110]:
```suggestion
    description = "Enterprise-class LDAP server";
```

**Comment by risicle** [comment]
`_389-ds-base`?

File: pkgs/servers/389-directory-server/default.nix [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    homepage = "https://www.port389.org/";
+    license = licenses.asl20;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.ners ];
```
Comment (SuperSandro2000) [inline pkgs/servers/389-directory-server/default.nix:114]: Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

**Comment by risicle** [comment]
The point I'm trying to make with the above is - do we not already have a package for this? Under `pkgs/servers/ldap/389/default.nix`

File: pkgs/servers/389-directory-server/default.nix (hunk 5 above)
Comment (ners) [reply pkgs/servers/389-directory-server/default.nix:114]: This was already done in [the other PR](https://github.com/NixOS/nixpkgs/pull/192649). :slightly_smiling_face:

**Comment by ners** [comment]
Nice catch, @risicle. I'll merge my package into the existing one, as it is more recent and builds the Rust components correctly.

**Comment by ners** [comment]
I've updated the existing package, as well as the PR description.

File: pkgs/servers/ldap/389/default.nix [hunk 6]
```diff
@@ -1,144 +1,117 @@
...
     repo = pname;
     rev = "${pname}-${version}";
-    sha256 = "sha256-aM1qo+yHrCFespPWHv2f25ooqQVCIZGaZS43dY6kiC4=";
+    sha256 = "sha256-GnntF0UaufDrgcM6FFFdwxwVoU9Hu2NXTW1A2lTb6T4=";
   };
 
-  nativeBuildInputs = [ autoreconfHook pkg-config doxygen ];
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = "${src}/src/Cargo.lock";
+  };
```
Comment (SuperSandro2000) [inline pkgs/servers/ldap/389/default.nix:49]: I think that is IFD and not allowed those ofborg fails.

**Comment by SuperSandro2000** [comment]
@ofborg build _389-ds-base

**Pull request merged**